src/legacy.py

```diff
@@ -0,0 +1,441 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# We ignore all lint errors in the legacy charm code as we've decided
+# to reuse them for the support of the legacy relations
+# flake8: noqa
+# pylint: skip-file
+# mypy: ignore-errors
+# fmt: off
+"""Legacy haproxy module."""
```

Comment:
We should be more explicit on the fact that we took the legacy implementation, and explain the reasoning.
Also eventually comment lines that we had to adapt